controllers/managedocs_controller.go

```diff
@@ -146,3 +156,32 @@ func (r *ManagedOCSReconciler) setDesiredStorageCluster(
 
 	return nil
 }
+
+func (r *ManagedOCSReconciler) updateComponents(req ctrl.Request) (v1.ManagedOCSComponents, error) {
+	var components v1.ManagedOCSComponents
+
+	// Checking the StorageCluster component.
+	var sc ocsv1.StorageCluster
+
+	scNamespaceName := types.NamespacedName{
+		Name:      storageClusterName,
+		Namespace: req.Namespace,
+	}
+
+	err := r.Get(r.ctx, scNamespaceName, &sc)
+	if err == nil {
+		if sc.Status.Phase == "Ready" {
+			components.StorageCluster.Status = v1.ComponentReady
+		} else {
+			components.StorageCluster.Status = v1.ComponentPending
+		}
+	} else if errors.IsNotFound(err) {
+		components.StorageCluster.Status = v1.ComponentNone
+	} else {
+		r.Log.Error(err, "error getting StorageCluster")
+		return components, err
```

Comment:
Why do we need to fail the reconcile in this case. 
Why not just specify the status as 'Unknown' and log the error?